test_ancestor_branch and test_branch_and_gc are flaky

Examples of `test_ancestor_branch`:

https://app.circleci.com/pipelines/github/neondatabase/neon/8032/workflows/f017e5f0-940d-41ce-bb71-7197930ceb17/jobs/82345
https://app.circleci.com/pipelines/github/neondatabase/neon/8032/workflows/c6c23a56-dc0b-41e7-96c8-112317be5fc4/jobs/82328/tests#failed-test-0

Logs are the almost same, here is the first one:

```
Exception:             Run ['/tmp/zenith/bin/neon_local', 'timeline', 'branch', '--branch-name', 'branch2', '--tenant-id', 'f87be4beaf274b768268ce637e189cf5', '--ancestor-branch-name', 'branch1', '--ancestor-start-lsn', '0/2C4C898'] failed:
              stdout: 
              stderr: command failed: error sending request for url (http://localhost:15004/v1/tenant/f87be4beaf274b768268ce637e189cf5/timeline): operation timed out

Caused by:
    operation timed out
subprocess.CalledProcessError: Command '['/tmp/zenith/bin/neon_local', 'timeline', 'branch', '--branch-name', 'branch2', '--tenant-id', 'f87be4beaf274b768268ce637e189cf5', '--ancestor-branch-name', 'branch1', '--ancestor-start-lsn', '0/2C4C898']' returned non-zero exit status 1.

The above exception was the direct cause of the following exception:
test_runner/batch_others/test_ancestor_branch.py:72: in test_ancestor_branch
    env.neon_cli.create_branch('branch2', 'branch1', tenant_id=tenant, ancestor_start_lsn=lsn_200)
test_runner/fixtures/neon_fixtures.py:1100: in create_branch
    res = self.raw_cli(cmd)
test_runner/fixtures/neon_fixtures.py:981: in raw_cli
    raise Exception(msg) from subprocess.CalledProcessError(res.returncode,
E   Exception:             Run ['/tmp/zenith/bin/neon_local', 'timeline', 'branch', '--branch-name', 'branch2', '--tenant-id', 'f87be4beaf274b768268ce637e189cf5', '--ancestor-branch-name', 'branch1', '--ancestor-start-lsn', '0/2C4C898'] failed:
E                 stdout: 
E                 stderr: command failed: error sending request for url (http://localhost:15004/v1/tenant/f87be4beaf274b768268ce637e189cf5/timeline): operation timed out
E   
E   Caused by:
E       operation timed out
```

--- a/test_runner/batch_others/test_branching.py
+++ b/test_runner/batch_others/test_branching.py
@@ -44,7 +44,7 @@ def test_branching_with_pgbench(neon_simple_env: NeonEnv,
         log.info(f"Start a pgbench workload on pg {connstr}")
 
         pg_bin.run_capture(['pgbench', '-i', f'-s{scale}', connstr])
-        pg_bin.run_capture(['pgbench', '-c10', '-T15', connstr])
+        pg_bin.run_capture(['pgbench', '-T15', connstr])
 
     env.neon_cli.create_branch('b0', tenant_id=tenant)
     pgs: List[Postgres] = []
@@ -54,11 +54,22 @@ def test_branching_with_pgbench(neon_simple_env: NeonEnv,
     threads.append(threading.Thread(target=run_pgbench, args=(pgs[0], ), daemon=True))
     threads[-1].start()
 
+    thread_limit = 4
+
     for i in range(n_branches):
         # random a delay between [0, 5]
         delay = random.random() * 5
         time.sleep(delay)
         log.info(f"Sleep {delay}s")
+
+        # If the number of concurrent threads exceeds a threshold,
+        # wait for all the threads to finish before spawning a new one.
+        # Because tests defined in `batch_others` are run concurrently in CI,
+        # we want to avoid the situation that one test exhausts resources for other tests.
+        if len(threads) >= thread_limit:
+            for thread in threads:
+                thread.join()
+            threads = []
 
         if ty == "cascade":
             env.neon_cli.create_branch('b{}'.format(i + 1), 'b{}'.format(i), tenant_id=tenant)

